src/main/java/ch/njol/skript/structures/StructAutoReload.java

```diff
@@ -0,0 +1,233 @@
+import org.jetbrains.annotations.Unmodifiable;
+import org.skriptlang.skript.lang.entry.EntryContainer;
+import org.skriptlang.skript.lang.entry.EntryData;
+import org.skriptlang.skript.lang.entry.EntryValidator;
+import org.skriptlang.skript.lang.script.Script;
+import org.skriptlang.skript.lang.script.ScriptData;
+import org.skriptlang.skript.lang.structure.Structure;
+
+import com.google.common.collect.Lists;
+
+@Name("Auto Reload")
+@Description({
+	"Place at the top of a script file to enable and configure automatic reloading of the script.",
+	"When the script is saved, Skript will automatically reload the script.",
+	"The config.sk node 'script loader thread size' must be set to a positive number for this to be enabled.",
+	"",
+	"available optional nodes:",
+	"\trecipients: The players to send reload messages to. Defaults to console.",
+	"\tpermission: The permission required to receive reload messages. 'recipients' will override this node.",
+})
+@Examples({
+	"auto reload",
+	"",
+	"auto reload:",
+	"\trecipients: \"SkriptDev\", \"61699b2e-d327-4a01-9f1e-0ea8c3f06bc6\" and \"Njol\"", // UUID is Dinnerbone's.
+	"\tpermission: \"skript.reloadnotify\"",
+})
+@Since("INSERT VERSION")
+public class StructAutoReload extends Structure {
+
+	public static final Priority PRIORITY = new Priority(10);
+
+	static {
+		EntryValidator entryValidator = EntryValidator.builder()
+			// Uses OfflinePlayer because this is determined at parse time. Runtime will make sure it's a Player.
+			.addEntryData(new EntryData<OfflinePlayer[]>("recipients", null, true) {
+
+				@Override
+				public OfflinePlayer @Nullable [] getValue(Node node) {
+					EntryNode entry = (EntryNode) node;
+					Literal<? extends OfflinePlayer> recipients = SkriptParser.parseLiteral(entry.getValue(), OfflinePlayer.class, ParseContext.DEFAULT);
+					return recipients.getArray();
+				}
+
+				@Override
+				public boolean canCreateWith(Node node) {
+					return node instanceof EntryNode;
+				}
+
+			})
+			.addEntry("permission", "skript.reloadnotify", true)
+			.build();
+		
+		Skript.registerStructure(StructAutoReload.class, entryValidator, "auto[matically] reload");
+	}
+
+	private Script script;
+	private Task task;
+
+	@Override
+	public boolean init(Literal<?> @NotNull [] arguments, int pattern, ParseResult result, EntryContainer container) {
+		try {
+			int threadSize = Integer.parseInt(SkriptConfig.scriptLoaderThreadSize.value());
+			if (threadSize <= 0)
+				throw new IllegalStateException();
+		} catch (IllegalStateException | NumberFormatException e) {
+			Skript.error(Language.get("log.auto reload.async required"));
+			return false;
+		}
+
+		OfflinePlayer[] recipients = null;
+		String permission = "skript.reloadnotify";
+
+		// Container can be null if the structure is simple.
+		if (container != null) {
+			recipients = container.get("recipients", OfflinePlayer[].class, false);
+			permission = container.get("permission", String.class, false);
+		}
```

Comment:
I could be wrong, but I had issues when I didnt have recipients, it threw an error... so shouldn't these be getOptional?
```suggestion
		if (container != null) {
			recipients = container.getOptional("recipients", OfflinePlayer[].class, false);
			permission = container.getOptional("permission", String.class, false);
		}
```
(I had to do this change just to get this to run)

Reply:
What's the error if you still have it? I can debug it to get the error too.
If the default value is null, it should return the default value of null with no error. getOptional is probably a better method to use.

Reply:
using this code:
```vb
auto reload:
	permission: "test.test"
```
error:
```
[12:36:29 ERROR]: #!#! 
[12:36:29 ERROR]: #!#! [Skript] Severe Error:
[12:36:29 ERROR]: #!#! Could not load test.sk
[12:36:29 ERROR]: #!#! 
[12:36:29 ERROR]: #!#! An unexpected error occurred with Skript. This issue is likely not your fault.
[12:36:29 ERROR]: #!#! You are using some plugins that alter how Skript works (addons).
[12:36:29 ERROR]: #!#! Full list of addons:
[12:36:29 ERROR]: #!#! SkBriggy v1.4.3 (https://github.com/ShaneBeee/SkBriggy)
[12:36:29 ERROR]: #!#! SkBee v3.8.1 (https://github.com/ShaneBeee/SkBee)
[12:36:29 ERROR]: #!#! skript-reflect v2.5.1 (https://github.com/SkriptLang/skript-reflect)
[12:36:29 ERROR]: #!#! CorePlugin v1.0.0 (https://github.com/ShaneBeee/CorePlugin)
[12:36:29 ERROR]: #!#! We could not identify related addons, it might also be a Skript issue.
[12:36:29 ERROR]: #!#! Try temporarily removing the listed plugins one by one to identify the cause.
[12:36:29 ERROR]: #!#! If removing a plugin resolves the issue, please report the problem to the plugin developer.
[12:36:29 ERROR]: #!#! 
[12:36:29 ERROR]: #!#! Stack trace:
[12:36:29 ERROR]: #!#! Caused by: java.lang.RuntimeException: Null value for asserted non-null value
[12:36:29 ERROR]: #!#!     at Skript-2.10.0.jar//org.skriptlang.skript.lang.entry.EntryContainer.get(EntryContainer.java:78)
[12:36:29 ERROR]: #!#!     at Skript-2.10.0.jar//ch.njol.skript.structures.StructAutoReload.init(StructAutoReload.java:113)
[12:36:29 ERROR]: #!#!     at Skript-2.10.0.jar//org.skriptlang.skript.lang.structure.Structure.init(Structure.java:124)
[12:36:29 ERROR]: #!#!     at Skript-2.10.0.jar//ch.njol.skript.lang.SkriptParser.parse(SkriptParser.java:241)
[12:36:29 ERROR]: #!#!     at Skript-2.10.0.jar//ch.njol.skript.lang.SkriptParser.parseStatic(SkriptParser.java:175)
[12:36:29 ERROR]: #!#!     at Skript-2.10.0.jar//org.skriptlang.skript.lang.structure.Structure.parse(Structure.java:220)
[12:36:29 ERROR]: #!#!     at Skript-2.10.0.jar//org.skriptlang.skript.lang.structure.Structure.parse(Structure.java:203)
[12:36:29 ERROR]: #!#!     at Skript-2.10.0.jar//ch.njol.skript.ScriptLoader.loadScript(ScriptLoader.java:664)
[12:36:29 ERROR]: #!#!     at Skript-2.10.0.jar//ch.njol.skript.ScriptLoader.lambda$loadScripts$4(ScriptLoader.java:476)
[12:36:29 ERROR]: #!#!     at Skript-2.10.0.jar//ch.njol.skript.ScriptLoader.lambda$makeFuture$2(ScriptLoader.java:393)
[12:36:29 ERROR]: #!#!     at Skript-2.10.0.jar//ch.njol.skript.ScriptLoader$AsyncLoaderThread.run(ScriptLoader.java:343)
[12:36:29 ERROR]: #!#! 
[12:36:29 ERROR]: #!#! Skript: 2.10.0 (unknown)
[12:36:29 ERROR]: #!#!     Flavor: selfbuilt-unknown
[12:36:29 ERROR]: #!#!     Date: unknown
[12:36:29 ERROR]: #!#! Bukkit: 1.21.4-R0.1-SNAPSHOT
[12:36:29 ERROR]: #!#! Minecraft: 1.21.4
[12:36:29 ERROR]: #!#! Java: 22.0.2 (OpenJDK 64-Bit Server VM 22.0.2+9)
[12:36:29 ERROR]: #!#! OS: Mac OS X aarch64 15.2
[12:36:29 ERROR]: #!#! 
[12:36:29 ERROR]: #!#! Server platform: Paper
[12:36:29 ERROR]: #!#! 
[12:36:29 ERROR]: #!#! Current node: auto reload: (test.sk, line 2)
[12:36:29 ERROR]: #!#! Current item: null
[12:36:29 ERROR]: #!#! Thread: Thread-7
[12:36:29 ERROR]: #!#! Language: english
[12:36:29 ERROR]: #!#! Link parse mode: DISABLED
[12:36:29 ERROR]: #!#! End of Error.
[12:36:29 ERROR]: #!#! 
[12:36:29 INFO]: Succes
```

Reply:
Ah, it's an API design issue. When using the `get` method and the default value is literally null, it'll still trigger the exception to throw. getOptional should really be wrapped in an Optional to differentiate a null value from a default value being null. Might make a new PR to deprecate the getOptional returning the object in the future to be an Optional.

Will change to getOptional to fix.

Reply:
The default boolean can't be true either, otherwise it'll also trip the exception.

Reply:
Yeah, you can't have `null` as a default value. Use getOptional in those cases, as the return value for get is explicitly not null